src/Orleans.BroadcastChannel/BroadcastChannel.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.BroadcastChannel.SubscriberTable;
+using Orleans.Providers;
+using Orleans.Runtime;
+
+namespace Orleans.BroadcastChannel
+{
+    public interface IBroadcastChannel<T>
+    {
+        Task Publish(T item);
+    }
+
+    internal class BroadcastChannel<T> : IBroadcastChannel<T>
+    {
+        private readonly InternalChannelId _channelId;
+        private readonly IGrainFactory _grainFactory;
+        private readonly ImplicitChannelSubscriberTable _subscriberTable;
+        private readonly bool _fireAndForgetDelivery;
+        private readonly ILogger _logger;
+
+        public BroadcastChannel(
+            InternalChannelId channelId,
+            IGrainFactory grainFactory,
+            ImplicitChannelSubscriberTable subscriberTable,
+            bool fireAndForgetDelivery,
+            ILoggerFactory loggerFactory)
+        {
+            _channelId = channelId;
+            _grainFactory = grainFactory;
+            _subscriberTable = subscriberTable;
+            _fireAndForgetDelivery = fireAndForgetDelivery;
+            _logger = loggerFactory.CreateLogger($"{nameof(BroadcastChannel<T>)}-{_channelId}");
```

Comment:
We should avoid unique logger names like this, since it can result in a memory leak. See https://github.com/dotnet/orleans/issues/6289 and https://github.com/dotnet/orleans/issues/6096